src/main/kotlin/widgets/Selector.kt

```diff
@@ -0,0 +1,178 @@
+package widgets
+
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.PublishSubject
+import javafx.application.Platform
+import javafx.beans.property.SimpleObjectProperty
+import javafx.geometry.Insets
+import javafx.geometry.Pos
+import javafx.scene.control.ComboBox
+import javafx.scene.layout.Priority
+import javafx.scene.paint.Color
+import tornadofx.*
+import java.util.*
+
+/**
+ * This class is used to make allow a user to select items from a filterable list
+ * and select which of those that are selected as the preferred, or default, item.
+ */
+
+class Selector(selectionData : List<DataSelectionInterface>,
+                        label : String,
+                        hint : String,
+                        private val colorAccent : Color,
+                        private val updateSelections : PublishSubject<DataSelectionInterface>,
+                        private val preferredSelection : PublishSubject<DataSelectionInterface>
+) : Fragment() {
+
+    init {
+        messages = ResourceBundle.getBundle("MyView")
+    }
+
+    private val compositeDisposable = CompositeDisposable()
+    private val viewModel = SelectorViewModel(updateSelections, preferredSelection)
+
+
+    private val chips = mutableListOf<Chip>()
+    private val selectionList = SelectionList(selectionData)
+    private val input = SimpleObjectProperty<String>()
+
+
+    private val colorNeutral = c(messages["UI_NEUTRAL"])
+    private val textFillNeutral = c(messages["UI_NEUTRALTEXT"])
```

Comment:
can we make this UI_NEUTRAL_TEXT?

Reply:
Done

Reply:
It _was_ UI_NEUTRAL_TEXT before **someone** changed it @cabenedick

Reply:
additionally, the widget shouldn't be dependent on app resources

Reply:
How are you supposed to use the resource folder to store colors, then? D:

Reply:
Passed those colors in as well as the accent color so that the widget is no longer dependent